src/import.rs

```diff
@@ -69,18 +69,18 @@ pub async fn import_many(
             let image = entries
                 .iter()
                 .enumerate()
-                .find(|(_, entry)| {
-                    entry
-                        .entry()
-                        .filename()
-                        .as_str()
-                        .map(|name| name == "icon.png" || name == "icon.jpg")
-                        .unwrap_or_default()
+                .map(|(index, entry)| {
+                    (index, entry.entry().filename().as_str().unwrap_or_default())
                 })
-                .map(|a| a.0);
-            let image = if let Some(index) = image {
+                .find(|(_, name)| *name == "icon.png" || *name == "icon.jpg");
+            let image = if let Some((index, name)) = image {
                 let res = read_vec(&reader, index).await?;
-                encode(&res)
+                let ending = name
```

Comment:
```suggestion
                let extension = name
```

the `.png` or `.jpg` part of a file name is called "extension" or "suffix", i would stay with that well known wording (also change at the other places)